lesson2/src/linked_list/listProblems.js

```diff
@@ -26,7 +53,22 @@ class ListProblems {
    pointers should be easier.
    */
   static kthToLast(list, k) {
-   // todo ✏️
+    let node = list.head;// used to loop through the list
+
+    let wantedIndex = 0; // will represent the k-th to end index from the beginning (length - k)
+
+    while (node) {
+      wantedIndex++;
+      node = node.next;
+    }
+
+    wantedIndex -= k; // get the index in the right order
```

Comment:
nice! and O(n), you traversed the list twice (one to get the length and the next time to get the kth to last) you can do a bit better if you use 2 pointers, the one you are using `let node = list.head;` and one that starts k elements after that one.. i went for this approach

```    // slow pointer traverses the list normally
    // fast pointer traverses the list starting k elements after.
    let slow = list.head;
    let fast = list.head;
    for (let i = 0; i < k; i++) {
      if (fast === null) {
        return -1;
      }
      fast = fast.next;
    }```

now i have 2 pointers k nodes apart.... not going to spoiler the whole solution 😉